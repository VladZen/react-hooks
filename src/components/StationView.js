import { startCase, isNull, isUndefined } from 'lodash'

import StationType from '@/types/Station'

import '@styles/components/station-view.scss'

const StationView = ({ station }) => {
  const humanizeValue = (key) => {
    const value = station[key]

    // process null
    if (isNull(value) || isUndefined(value)) return 'Not detected'

    // process enum
    if ([0, 1].includes(value)) return (
      <span className={`text-${value === 1 ? 'positive' : 'negative'}`}>
        { value === 1 ? 'Yes' : 'No'}
      </span>
    )

    // process date
    if (key === 'lastconnect') return new Date(value).toLocaleString()

    // process position
    if (key === 'position') return (
      <a
        className='link-decorated'
        href={`http://www.google.com/maps/place/${value}`}
        target='blank'
        title='Open in new tab'
      >
        Show on the map
      </a>
    )

    // default
    return value
  }

  return (
    <dl className='station-attributes'>
      {
        Object.keys(station).map((key) => (
          <div className='station-attributes__item' key={ key }>
            <dt className='station-attributes__item__label'>{ startCase(key) }</dt>
            <dd className='station-attributes__item__value'>{ humanizeValue(key) }</dd>
          </div>
        ))
      }
    </dl>
  )
}

StationView.propTypes = {
  station: StationType
}

export default StationView
