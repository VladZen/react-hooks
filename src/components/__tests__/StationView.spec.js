import { unmountComponentAtNode, render } from 'react-dom'
import { act } from 'react-dom/test-utils'
import StationView from '@/components/StationView'

let container = null
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

const mockStation = {
  'station_ID': 102,
  'custom_evse_id': null,
  'location_ID': 102,
  'seller_ID': 27,
  'name': 'Endesa 2',
  'connected': 1,
  'position': '60.868623,26.702901',
  'available': 0,
  'lastconnect': '2019-02-22T13:36:12.000Z',
  'roaming_identifier_cpo': 'FI*001'
}

describe('StationView component', () => {
  it('should be rendered properly', () => {
    act(() => {
      render(
        (<StationView station={ mockStation } />),
        container
      )
    })

    expect(container).toMatchSnapshot()
  })
})
