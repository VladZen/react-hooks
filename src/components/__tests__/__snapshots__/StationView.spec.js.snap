// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StationView component should be rendered properly 1`] = `
<div>
  <dl
    class="station-attributes"
  >
    <div
      class="station-attributes__item"
    >
      <dt
        class="station-attributes__item__label"
      >
        Station ID
      </dt>
      <dd
        class="station-attributes__item__value"
      >
        102
      </dd>
    </div>
    <div
      class="station-attributes__item"
    >
      <dt
        class="station-attributes__item__label"
      >
        Custom Evse Id
      </dt>
      <dd
        class="station-attributes__item__value"
      >
        Not detected
      </dd>
    </div>
    <div
      class="station-attributes__item"
    >
      <dt
        class="station-attributes__item__label"
      >
        Location ID
      </dt>
      <dd
        class="station-attributes__item__value"
      >
        102
      </dd>
    </div>
    <div
      class="station-attributes__item"
    >
      <dt
        class="station-attributes__item__label"
      >
        Seller ID
      </dt>
      <dd
        class="station-attributes__item__value"
      >
        27
      </dd>
    </div>
    <div
      class="station-attributes__item"
    >
      <dt
        class="station-attributes__item__label"
      >
        Name
      </dt>
      <dd
        class="station-attributes__item__value"
      >
        Endesa 2
      </dd>
    </div>
    <div
      class="station-attributes__item"
    >
      <dt
        class="station-attributes__item__label"
      >
        Connected
      </dt>
      <dd
        class="station-attributes__item__value"
      >
        <span
          class="text-positive"
        >
          Yes
        </span>
      </dd>
    </div>
    <div
      class="station-attributes__item"
    >
      <dt
        class="station-attributes__item__label"
      >
        Position
      </dt>
      <dd
        class="station-attributes__item__value"
      >
        <a
          class="link-decorated"
          href="http://www.google.com/maps/place/60.868623,26.702901"
          target="blank"
          title="Open in new tab"
        >
          Show on the map
        </a>
      </dd>
    </div>
    <div
      class="station-attributes__item"
    >
      <dt
        class="station-attributes__item__label"
      >
        Available
      </dt>
      <dd
        class="station-attributes__item__value"
      >
        <span
          class="text-negative"
        >
          No
        </span>
      </dd>
    </div>
    <div
      class="station-attributes__item"
    >
      <dt
        class="station-attributes__item__label"
      >
        Lastconnect
      </dt>
      <dd
        class="station-attributes__item__value"
      >
        2/22/2019, 4:36:12 PM
      </dd>
    </div>
    <div
      class="station-attributes__item"
    >
      <dt
        class="station-attributes__item__label"
      >
        Roaming Identifier Cpo
      </dt>
      <dd
        class="station-attributes__item__value"
      >
        FI*001
      </dd>
    </div>
  </dl>
</div>
`;
